pool/src/types.rs

```diff
@@ -88,17 +89,38 @@ pub enum PoolError {
         /// The spent output
         spent_output: Commitment
     },
+    /// Attempt to spend a coinbase output before it matures (1000 blocks?)
+    ImmatureCoinbase{
+        /// The block header of the block containing the coinbase output
+        header: block::BlockHeader,
+        /// The unspent coinbase output
+        output: Commitment,
+    },
     /// An orphan successfully added to the orphans set
     OrphanTransaction,
+    /// TODO - wip, just getting imports working, remove this and use more specific errors
+    GenericPoolError,
+    /// TODO - is this the right level of abstraction for pool errors?
+    OutputNotFound,
```

Comment:
Alternatively - do we just want a more generic `ChainError` here to represent something going wrong with accessing the chain?

Reply:
I'd stick with the more specific errors as long as there are only a few of them.